libraries/networking/src/udt/SendQueue.cpp

```diff
@@ -204,144 +225,159 @@ void SendQueue::run() {
     _isRunning = true;
     
     while (_isRunning) {
-        // Record timing
-        _lastSendTimestamp = high_resolution_clock::now();
+        // Record how long the loop takes to execute
+        auto loopStartTimestamp = clock::now();
         
-        bool naksEmpty = true; // used at the end of processing to see if we should wait for NAKs
+        bool sentAPacket = maybeResendPacket();
+        bool flowWindowFull = false;
         
-        bool resentPacket = false;
+        // if we didn't find a packet to re-send AND we think we can fit a new packet on the wire
+        // (this is according to the current flow window size) then we send out a new packet
+        if (!sentAPacket) {
+            flowWindowFull = (seqlen(SequenceNumber { (uint32_t) _lastACKSequenceNumber }, _currentSequenceNumber) >
+                              _flowWindowSize);
+            sentAPacket = maybeSendNewPacket();
+        }
+        
+        // Keep track of how long the flow window has been full for
+        if (flowWindowFull && !_flowWindowWasFull) {
+            _flowWindowFullSince = loopStartTimestamp;
+        }
+        _flowWindowWasFull = flowWindowFull;
```

Comment:
Well that's my understanding, since some packet got lost, the wire is not `full`, there is space for all the NAKed packets so I consider the flow window is not full.
For me a true full flow window is when you nak list is empty, do you agree?